src/Instrumentation/ExtRdKafka/src/ExtRdKafkaInstrumentation.php

```diff
@@ -0,0 +1,131 @@
+            TraceAttributes::SCHEMA_URL,
+        );
+
+        // Start root span and propagate parent if it exists in headers, for each message consumed
+        self::addConsumeHooks($instrumentation);
+        // End root span on offset commit
+        self::addCommitHooks('commit');
+        self::addCommitHooks('commitAsync');
+        // Context propagation for outbound messages
+        self::addProductionHooks();
+    }
+
+    private static function addCommitHooks($functionName)
+    {
+        hook(
+            KafkaConsumer::class,
+            $functionName,
+            null,
+            static function () {
+                $scope = Context::storage()->scope();
+                if ($scope === null) {
+                    return;
+                }
+                $scope->detach();
+                $span = Span::fromContext($scope->context());
+                $span->end();
+            }
+        );
+    }
+
+    private static function addProductionHooks()
+    {
+        hook(
+            ProducerTopic::class,
+            'producev',
+            static function (
+                ProducerTopic $exchange,
+                array $params
+            ): array {
+                // Headers are the 5th argument for the producev function
+                $carrier = [];
+                TraceContextPropagator::getInstance()->inject($carrier);
+                $propagator = Globals::propagator();
+                $propagator->inject($carrier);
+
+                if (array_key_exists('traceparent', $carrier)) {
+                    $params[4]['traceparent'] = $carrier['traceparent'];
+                }
+
+                return $params;
+            }
+        );
+    }
+
+    private static function addConsumeHooks($instrumentation)
+    {
+        hook(
+            KafkaConsumer::class,
+            'consume',
+            null,
+            static function (
+                ?KafkaConsumer $exchange,
+                array $params,
+                ?Message $message,
+                ?Throwable $exception
+            ) use ($instrumentation) : void {
+                // This is to ensure that there is data. Packages periodically poll this method in order to
+                // determine if there is a message there. If there is not, we don't want to create a span.
+                if (!$message instanceof Message || $message->err !== RD_KAFKA_RESP_ERR_NO_ERROR) {
+                    return;
+                }
+
+                $builder = $instrumentation
+                    ->tracer()
+                    // @phan-suppress-next-line PhanTypeSuspiciousStringExpression - Doesn't seem to know this has to be a string
+                    ->spanBuilder('Consumer: ' . $message->topic_name)
+                    ->setSpanKind(SpanKind::KIND_CONSUMER)
+                    ->setAttribute(TraceAttributes::MESSAGING_SYSTEM, 'kafka')
+                    ->setAttribute(TraceAttributes::MESSAGING_OPERATION, 'consume')
```

Comment:
note that messaging.operation has some well-known values listed in https://opentelemetry.io/docs/specs/semconv/attributes-registry/messaging/#messaging-attributes - maybe `receive` or `process` is more appropriate here?

is there anything else of value in https://opentelemetry.io/docs/specs/semconv/attributes-registry/messaging/#messaging-kafka-attributes that you could add?

Reply:
Changed the attribute to `process`. Also added a couple more attributes